### PR TITLE
Release of optimade-validator-action v2

### PR DESCRIPTION
**Pull request by ml-evs**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 
-All minor and patch updates to v1 will be folded into the `v1` tag, so that using the action `@v1` is recommended, since it results in using the latest v1.minor.patch.
+All minor and patch updates to v2 will be folded into the `v2` tag, so that using the action `@v2` is recommended, since it results in using the latest v2.minor.patch.
 
 Latest versions:
 
 | Used tag | Effective version |
 | :---: | :---: |
+| `v2` | [`v2.0.0`](https://github.com/Materials-Consortia/optimade-validator-action/releases/tag/v2.0.0)
 | `v1` | [`v1.1.0`](https://github.com/Materials-Consortia/optimade-validator-action/releases/tag/v1.1.0)
```
Comment (CasperWA) [inline README.md:14]: I know you didn't touch this, but I forgot to update this with a previous release.
```suggestion
| `v1` | [`v1.2.0`](https://github.com/Materials-Consortia/optimade-validator-action/releases/tag/v1.2.0)
```

File: README.md [hunk 2]
```diff
@@ -43,19 +44,19 @@ with:
...
+| Input | Description | Usage | Default | Action version |
+| :---: |    :---     | :---: |  :---:  |      :---:     |
+| `all_versioned_paths` | Whether to test all possible versioned base URLs:<br><br>/vMAJOR<br>/vMAJOR.MINOR<br>/vMAJOR.MINOR.PATCH<br><br>If this is `'true'`, the input `'path'` MUST exempt the version part (e.g., `'/optimade'` instead of `'/optimade/v1'`).<br>If this is `'false'`, the input `'path'` MUST include the version part (e.g., `'/optimade/v1'` instead of `'/optimade'`) | Optional | `false` | `v1+`
+| `as_type` | Validate the request URL with the provided type, rather than scanning the entire implementation<br>Example values: 'structures', 'reference'. For a full list of values see `optimade-python-tools`. | Optional | - | `v1+`
+| `domain` | Domain for the OPTIMADE URL (defaults to the GitHub Actions runner host) | Optional | `gh_actions_host` | `v1+`
+| `fail_fast` | Whether or not to exit and return a non-zero error code on first failure. | Optional | `false` | `v2+`
+| `index` | Whether or not this is an index meta-database | Optional | `false` | `v2+`
```
Comment (CasperWA) [inline README.md:53]: This was always there :)
```suggestion
| `index` | Whether or not this is an index meta-database | Optional | `false` | `v1+`
```

File: README.md [hunk 3]
```diff
@@ -105,3 +106,14 @@ Another is the relative or absolute path to the `.bats` files directory or a sin
...
+
+To keep it simple, this overview will only consider the major versions of this Action:
+
+| Action version | `optimade` package versions | Supported OPTIMADE API specification version(s) |
+| :---: | :---: | :---: |
+| [`v2`](https://github.com/Materials-Consortia/optimade-validator-action/releases/tag/v2.0.0) | [`v0.10.0`](https://github.com/Materials-Consortia/optimade-python-tools/releases/v0.10.0)+ | [`v1.0.0`](https://github.com/Materials-Consortia/OPTIMADE/blob/v1.0.0/optimade.rst) |
+| [`v1`](https://github.com/Materials-Consortia/optimade-validator-action/releases/tag/v1.2.0) | [`v0.7.0`](https://github.com/Materials-Consortia/optimade-python-tools/releases/v0.7.0) - [`v0.9.8`](https://github.com/Materials-Consortia/optimade-python-tools/releases/v0.9.8) | [`v0.10.0`](https://github.com/Materials-Consortia/OPTIMADE/blob/v0.10.0/optimade.md) - [`v1.0.0`](https://github.com/Materials-Consortia/OPTIMADE/blob/v1.0.0/optimade.rst) |
```
Comment (ml-evs) [inline README.md:119]:
```suggestion
| [`v1`](https://github.com/Materials-Consortia/optimade-validator-action/releases/tag/v1.2.0) | [`v0.7.0`](https://github.com/Materials-Consortia/optimade-python-tools/releases/v0.7.0) - [`v0.9.8`](https://github.com/Materials-Consortia/optimade-python-tools/releases/v0.9.8) | [`v0.10.0`](https://github.com/Materials-Consortia/OPTIMADE/blob/v0.10.0/optimade.md) - [`v1.0.0-rc2`](https://github.com/Materials-Consortia/OPTIMADE/blob/v1.0.0-rc2/optimade.rst) |
```


Think this suggestion is valid, as we definitely don't *currently* validate 1.0 compliant implementations correctly, but hopefully we were pretty close for rc2

Comment (ml-evs) [reply README.md:119]: (Mostly because of the issues about model strictness)

Comment (ml-evs) [reply README.md:119]: and of course there is the baked in assumption that v2 will eventually work for 1.0, which hopefully isn't that far away

Comment (CasperWA) [reply README.md:119]: Fair enough :)

File: tests/test_validator_version.bats [hunk 4]
```diff
@@ -30,8 +30,8 @@ load 'test_fixtures'
     assert_output "run_validator: ${TEST_FINAL_RUN_VALIDATOR}"
 }
 
-@test "validator_version='master'" {
-    export INPUT_VALIDATOR_VERSION=master
+@test "validator_version='latest'" {
+    export INPUT_VALIDATOR_VERSION=latest
```
Comment (CasperWA) [inline tests/test_validator_version.bats:34]: This shouldn't be changed as it's a test for whether it works to install a branch.
```suggestion
@test "validator_version='master'" {
    export INPUT_VALIDATOR_VERSION=master
```

**Pull request merged**